dev/CommonStyles/TestUI/CornerRadiusPage.xaml

```diff
@@ -0,0 +1,684 @@
+            <Grid.Resources>
+                <Style x:Key="SideHeader" TargetType="TextBlock">
+                    <Setter Property="HorizontalAlignment" Value="Right"/>
+                    <Setter Property="Grid.Column" Value="0"/>
+                </Style>
+                <Style x:Key="LeftColumn" TargetType="Grid">
+                    <Setter Property="HorizontalAlignment" Value="Left"/>
+                    <Setter Property="VerticalAlignment" Value="Top"/>
+                    <Setter Property="Grid.Column" Value="1"/>
+                </Style>
+                <Style x:Key="RightColumn" TargetType="Grid">
+                    <Setter Property="HorizontalAlignment" Value="Left"/>
+                    <Setter Property="VerticalAlignment" Value="Top"/>
+                    <Setter Property="Grid.Column" Value="2"/>
+                </Style>
+            </Grid.Resources>
+            <Grid.ColumnDefinitions>
+                <ColumnDefinition Width="150"/>
+                <ColumnDefinition Width="Auto"/>
+                <ColumnDefinition Width="Auto"/>
+            </Grid.ColumnDefinitions>
+            <Grid.RowDefinitions>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
+                <RowDefinition/>
```

Comment:
Oog.  This giant pile of row definitions is kind of unsightly.  I might suggest something like [RelativePanel](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.relativepanel) as an alternative.  Then you could do something like this:

```
            <TextBlock x:Name="ListBoxTextBlock" Text="ListBox" Style="{StaticResource SideHeader}"/>
            <Grid x:Name="ListBoxLeftGrid" RelativePanel.RightOf="ListBoxTextBlock" Style="{StaticResource LeftColumn}">
                <ListBox>
                    <ListBoxItem Content="Item 1"/>
                    <ListBoxItem Content="Item 2"/>
                    <ListBoxItem Content="Item 3"/>
                </ListBox>
            </Grid>
            <Grid x:Name="ListBoxRightGrid" RelativePanel.RightOf="ListBoxLeftGrid" Style="{StaticResource RightColumn}">
                <ListBox CornerRadius="6">
                    <ListBoxItem Content="Item 1"/>
                    <ListBoxItem Content="Item 2"/>
                    <ListBoxItem Content="Item 3"/>
                </ListBox>
            </Grid>

            <TextBlock x:Name="ComboBoxTextBlock" RelativePanel.Below="ListBoxTextBlock" Text="ComboBox" Style="{StaticResource SideHeader}"/>
            <Grid x:Name="ComboBoxLeftGrid" RelativePanel.Below="ListBoxLeftGrid" Style="{StaticResource LeftColumn}">
                <ComboBox>
                    <ComboBoxItem Content="Item 1"/>
                    <ComboBoxItem Content="Item 2"/>
                    <ComboBoxItem Content="Item 3"/>
                </ComboBox>
            </Grid>
<!-- etc. -->
```

This way you can still keep things ordered how you want, but you don't need a huge pile of magic numbers that might need to be updated *en masse* if something needed to be inserted in the middle.

Of course, if you aren't planning on maintaining this file (I don't know if this is just a one-off), then this wouldn't be necessary.

Reply:
You could also potentially set the RelativePanel attached properties in code-behind to even further reduce the amount of maintenance required, if the elements in XAML are defined in a set order.

Reply:
I haven't used RelativePanel before. It looks like an interesting solution. I like how it handles the left/right column... I'm not as big of a fan of needing all the Below references.

This isn't the only test page that has a mountain of RowDefinitions to accomplish a layout like this. My actual plan for these is to use [GridLayout](https://github.com/microsoft/microsoft-ui-xaml/issues/696), which transforms this:

```
<TextBlock Grid.Row="9" Text="DatePicker" Style="{StaticResource SideHeader}"/> 
<Grid Grid.Row="9" Style="{StaticResource LeftColumn}"> 
  <DatePicker /> 
</Grid> 
<Grid Grid.Row="9" Style="{StaticResource RightColumn}"> 
  <DatePicker CornerRadius="6"/> 
</Grid>
```
Into this:
```
<TextBlock Text="DatePicker"/> 
<DatePicker /> 
<DatePicker CornerRadius="6"/> 
```
No joke, it gets that simple. By setting up AutoColumns and AutoFlow correctly I can completely remove the need for these containers and grid definitions on the items and on the parent. I'm planning on converting the TeachingTip tests as a demonstration, because it has a similar problem.

I don't really want to shift this into code-behind as is because I think we can solve this in markup with a descriptive layout.

What I would be interested in moving into code-behind would be the whole pattern of "blast out every single control and have a before/after layout". I have a similar test page for control Header, for example. So I'd love to figure out a way to make control-wide tests like that in an automated way (similar to the state tester @teaP has), but I didn't want to take on for this change.

Reply:
If this eventually became that simple, I'd love that!  If you've got a plan to get there eventually, I'm fine with things as they are now; I was just a bit scared by the maintainability requirements implied in having dozens of hard-coded grid row numbers.